src/components/walletIntegration/AddAssetButton.tsx

```diff
@@ -31,8 +33,10 @@ function AddAssetButton({ asset_id, short_name, image_url, name, onCompletion =
         setIsLoading(false);
     }
 
+    const isWalletConnectActuallyConnected = connectedWallet === "WalletConnect" ? Boolean(connectedWallet === "WalletConnect" && walletConnectSelectedSession) : true;
+
     // Only display button if it's supported by the active wallet
-    if (walletName === "Goby" ||  walletName === "WalletConnect") {
+    if ((walletName === "Goby" ||  walletName === "WalletConnect") && connectedWallet && isWalletConnectActuallyConnected) {
```

Comment:
you forgot `isWalletConnectActuallyActuallyConnectedOrJustPretendingToBeConnected` (joke, ignore)